app/models/manageiq/providers/google/cloud_manager/event_catcher/runner.rb

```diff
@@ -0,0 +1,74 @@
+class ManageIQ::Providers::Google::CloudManager::EventCatcher::Runner <
+  ManageIQ::Providers::BaseManager::EventCatcher::Runner
+
+  # Request the event monitor to stop running.
+  def stop_event_monitor
+    stop_running
```

Comment:
you are right about, that we need this logic here.

But I think you can get along with less methods by just using a private instance variable
```ruby
def stop_event_monitor
  @stop_event_monitor = true
end

def monitor_events
  while not @stop_event_montior
  end
end
```

But actually all this logic shouldnt be here in the first place, but for now its required 😢